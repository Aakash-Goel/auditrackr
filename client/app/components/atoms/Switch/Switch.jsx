import React from 'react';
import { object } from 'prop-types';

import { withStyles } from '@material-ui/styles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

import switchStyles from './Switch.style';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  formControlLabelProps: object,
  switchProps: object,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  formControlLabelProps: {},
  switchProps: {},
};

const CustomSwitch = ({ ...props }) => {
  const { formControlLabelProps, switchProps } = props;

  return (
    <FormControlLabel
      {...formControlLabelProps}
      control={<Switch color="primary" {...switchProps} />}
    />
  );
};

CustomSwitch.propTypes = propTypes;
CustomSwitch.defaultProps = defaultProps;

export default withStyles(switchStyles)(CustomSwitch);
