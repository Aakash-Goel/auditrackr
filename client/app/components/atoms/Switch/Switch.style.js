// import { primaryColor } from '../../../styles/theme/muiKit';

const switchStyles = {
  disabled: {
    color: `rgba(0, 0, 0, 0.45)`,
  },
};

export default switchStyles;
