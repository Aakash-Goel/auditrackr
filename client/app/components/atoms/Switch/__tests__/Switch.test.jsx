import React from 'react';
import TestRenderer from 'react-test-renderer';

import Switch from '../Switch';

describe('<Switch/>', () => {
  it('renders correctly', () => {
    const switchWrapper = TestRenderer.create(<Switch />).toJSON();
    expect(switchWrapper).toMatchSnapshot();
  });
});
