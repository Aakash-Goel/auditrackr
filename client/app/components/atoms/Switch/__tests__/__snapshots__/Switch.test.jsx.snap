// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> renders correctly 1`] = `
<label
  className="MuiFormControlLabel-root-2"
>
  <span
    className="MuiSwitch-root-8"
  >
    <span
      className="MuiButtonBase-root-27 MuiIconButton-root-21 MuiPrivateSwitchBase-root-17 MuiSwitch-switchBase-11 MuiSwitch-colorPrimary-13"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-26"
      >
        <span
          className="MuiSwitch-icon-9"
        />
        <input
          className="MuiPrivateSwitchBase-input-20"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-66"
      />
    </span>
    <span
      className="MuiSwitch-bar-16"
    />
  </span>
  <span
    className="MuiTypography-root-30 MuiTypography-body1-39 MuiFormControlLabel-label-7"
  />
</label>
`;
